subsys/net/ip/tcp.c

```diff
@@ -2921,6 +2932,13 @@ static enum net_verdict tcp_in(struct tcp *conn, struct net_pkt *pkt)
 			} else {
 				verdict = NET_OK;
 			}
+
+#if defined(CONFIG_NET_TCP_IPV6_ND_REACHABILITY_HINT)
```

Comment:
Maybe instead of ifdef-slicing the code here and below, just provide a dummy implementation of `tcp_nbr_reachability_hint()` in case feature is disabled (like we do for Keep-alive feature, see `keep_alive_timer_init()` for instance)?